service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderControllerTest.java

```diff
@@ -0,0 +1,213 @@
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C21;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readBytes;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(C21OrderController.class)
+@OverrideAutoConfiguration(enabled = true)
+class C21OrderControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final String C21_DOCUMENT_TITLE = C21.getDocumentTitle();
+
+    @MockBean
+    private CreateC21OrderService createC21OrderService;
+
+    @MockBean
+    private DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    void aboutToStartShouldReturnErrorsWhenFamilymanNumberIsNotProvided() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of("data", "some data"))
+                .build())
+            .build();
+
+        MvcResult response = makeRequest(request, "about-to-start");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = objectMapper.readValue(
+            response.getResponse().getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).containsOnlyOnce("Enter Familyman case number");
+    }
+
+    @Test
+    void midEventShouldGenerateC21OrderDocument() throws Exception {
+        byte[] pdf = {1, 2, 3, 4, 5};
+        DocmosisDocument docmosisDocument = new DocmosisDocument(C21_DOCUMENT_TITLE, pdf);
+        Document document = document();
+
+        CallbackRequest caseDetails = mapper.readValue(readBytes("core-case-data-store-api/callback-request.json"),
+            CallbackRequest.class);
+
+        CaseData caseData = mapper.convertValue(caseDetails.getCaseDetails().getData(), CaseData.class);
+
+        Map<String, Object> templateData = createTemplatePlaceholders();
+
+        given(createC21OrderService.getC21OrderTemplateData(caseData))
+            .willReturn(templateData);
+        given(docmosisDocumentGeneratorService.generateDocmosisDocument(templateData, C21))
+            .willReturn(docmosisDocument);
+        given(uploadDocumentService.uploadPDF(USER_ID, AUTH_TOKEN, pdf, C21_DOCUMENT_TITLE + "_1.pdf"))
+            .willReturn(document);
```

Comment:
I don't actually care about most of this stuff, muddies the test. I want tests to be short and consinse, consider extratcing to private methods or using @Before blocks. When I read a test I wanna see, cool ok I am hitting this method with this stuff and yep I am asserting this thing is as expected. I don't care about a lot of the mockito mocks, I don't care where you are getting caseDetails from.... etc